tracer/src/Datadog.Trace/Debugger/LiveDebugger.cs

```diff
@@ -280,6 +284,89 @@ internal void UpdateAddedProbeInstrumentations(IReadOnlyList<ProbeDefinition> ad
             }
         }
 
+        private static string[] SafeParseSignature(string signature)
+        {
+            if (string.IsNullOrEmpty(signature))
+            {
+                return null;
+            }
+
+            try
+            {
+                return ParseSignature(signature);
+            }
+            catch (Exception e)
+            {
+                Log.Error(e, "Error parsing the signature {Signature}. Fall back to empty signature", signature);
+                return null;
+            }
+        }
+
+        private static string[] ParseSignature(string signature)
+        {
+            if (signature.StartsWith("(") && signature.EndsWith(")"))
+            {
+                signature = signature.Substring(1, signature.Length - 2);
+            }
+
+            var types = new List<string>();
+            var initialCapacity = 32;
+            var currentType = StringBuilderCache.Acquire(initialCapacity);
+            var stack = new Stack<char>();
```

Comment:
I don't think this needs to be a stack, I think a simple `int depth` would suffice

Reply:
https://github.com/DataDog/dd-trace-dotnet/pull/5829#discussion_r1700756550